File: src/main/java/io/github/mg138/tsbook/config/item/element/UnidentifiedSetting.java
package io.github.mg138.tsbook.config.item.element;

import org.bukkit.Material;
import org.bukkit.configuration.file.YamlConfiguration;

import java.util.List;

public class UnidentifiedSetting extends ItemSetting {
    public final List<String> iden;

    public UnidentifiedSetting(YamlConfiguration setting) {
        this(
                new SimpleItemSetting(setting),
                setting.getStringList("ITEMS")
        );
    }

    public UnidentifiedSetting(String id, String item_type, Material material, Integer model, String name, List<String> lore, List<String> iden) {
        this(new SimpleItemSetting(id, item_type, material, model, name, lore), iden);
    }

    public UnidentifiedSetting(ItemSetting setting, List<String> iden) {
        super(setting);
        this.iden = iden;
    }
}

File: src/main/kotlin/io/github/mg138/tsbook/config/item/element/UnidentifiedSetting.kt
package io.github.mg138.tsbook.config.item.element

class UnidentifiedSetting(setting: ItemSetting, val iden: List<String>) : ItemSetting(setting) {
    /*
    constructor(setting: YamlConfiguration) : this(
        ItemSetting(setting, Identifier.PresetKey.unid),
        setting.getStringList("ITEMS")
    )
    */
}